terasoluna-gfw-common-libraries/terasoluna-gfw-common/src/main/java/org/terasoluna/gfw/common/codelist/i18n/SimpleI18nCodeList.java

```diff
@@ -226,14 +232,54 @@
 
     /**
      * <p>
-     * returns row of codelist table.
+     * Returns a codelist as map for the {@link org.springframework.context.i18n.LocaleContextHolder}'s locale.
+     * </p>
+     * @see org.terasoluna.gfw.common.codelist.CodeList#asMap()
+     */
+    @Override
+    public Map<String, String> asMap() {
+        return asMap(LocaleContextHolder.getLocale());
+    }
+
+    /**
+     * <p>
+     * returns row of codelist table.<br>
+     * if there is no codelist for the locale, throws an {@link IllegalStateException}.
      * </p>
      * @see org.terasoluna.gfw.common.codelist.i18n.I18nCodeList#asMap(java.util.Locale)
      */
     @Override
     public Map<String, String> asMap(Locale locale) {
         Assert.notNull(locale, "locale is null");
-        return codeListTable.row(locale);
+
+        if (codeListTable.containsRow(locale)) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Found codelist for specified locale '{}'.",
+                        locale);
+            }
+            return codeListTable.row(locale);
+        }
+
+        Locale langOnlyLocale = new Locale(locale.getLanguage());
+        if (codeListTable.containsRow(langOnlyLocale)) {
+            if (logger.isDebugEnabled()) {
+                logger.debug(
+                        "Found codelist for specified locale '{}' (language only).",
+                        locale);
+            }
+            return codeListTable.row(langOnlyLocale);
+        }
+
+        if (locale.equals(fallbackTo)) {
+            throw new IllegalStateException("No codelist found for specified locale and fail to fallback");
+        }
```

Comment:
We talked about some issues of the CodeList first and you said when the CodeList failed to fallback we should throw an exception instead of returning empty map or `null`.
I think that is a relevant specification.
But will you change the opinion?

Reply:
@btkitsunedukam 
Sorry, Yes. I noticed that throwing exceptions would caused problems with `CodeListInterceptor` and View Rendering. As a CodeList alone, it is correct to throw an exception.

Reply:
@btkitsunedukam 
Probably, my original opinion is that it's better to assert whether or not exist code list for `fallbackTo` on `afterPropertiesSet`. Not good to assert on request processing.
How do you think?

Reply:
@yoshikawaa
I think it is better solution.
I will add the assertion to `afterPropertiesSet` and remove throwing `IllegalStateException` from `asMap` because existing `fallbackTo` was guaranteed.